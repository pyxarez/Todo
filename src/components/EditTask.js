import React from 'react';

import './EditTask.res/style.css';

EditTask.propTypes = {
  onSave: React.PropTypes.func.isRequired,
  onCancel: React.PropTypes.func.isRequired,
  children: React.PropTypes.oneOfType(
    [
      React.PropTypes.arrayOf(React.PropTypes.element),
      React.PropTypes.bool,
      React.PropTypes.object
    ]
  ),
}

export default function EditTask({ onSave, onCancel, children }) {
  return (
    <div className="my-editTask-component">
      <div className="buttons-container">
        <button onClick={onSave}>Save changes</button>
        <button onClick={onCancel}>Cancel</button>
      </div>
      {children}
    </div>
  );
}
