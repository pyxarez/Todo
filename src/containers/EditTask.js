import React, { Component, PropTypes } from 'react';
import { browserHistory } from 'react-router';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { validateInput } from '../utils/helpers';

import { saveTaskChanges } from '../store/actions/TaskListActions';

import EditTask from '../components/EditTask';

export class EditTaskContainer extends Component {
  static propTypes = {
    task: PropTypes.shape({
      id: PropTypes.node.isRequired,
      title: PropTypes.string.isRequired,
      isDone: PropTypes.bool.isRequired,
      description: PropTypes.string.isRequired,
    }),
    categoryId: PropTypes.number.isRequired,
    saveTaskChanges: PropTypes.func.isRequired,
  }

  onSave = () => {
    const title = this.titleInput.value;
    const isDone = this.doneCheckBox.checked;
    const description = this.taskDescription.value;

    if (!validateInput(title)) {
      alert('Enter correct task title');
      return;
    }

    const options = {
      categoryId: this.props.categoryId,
      id: this.props.task.id,
      title,
      isDone,
      description
    };

    this.props.saveTaskChanges(options);
    browserHistory.goBack();
  }

  handleCancelClick = () => {
    browserHistory.goBack();
  }

  render() {
    const {
      title,
      isDone,
      description
    } = this.props.task;

    return (
      <EditTask onSave={this.onSave} onCancel={this.handleCancelClick}>
        <div>
          <input ref={(titleInput) => {this.titleInput = titleInput}} type="text" defaultValue={title}/>
          <input ref={(doneCheckBox) => {this.doneCheckBox = doneCheckBox}} type="checkbox" defaultChecked={isDone}/>
        </div>
        <textarea
          ref={(taskDescription) => {this.taskDescription = taskDescription}}
          defaultValue={description}
          className="description"
          name="description"
          cols="30"
          rows="30" />
      </EditTask>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    saveTaskChanges: bindActionCreators(saveTaskChanges, dispatch),
  }
}

export default connect(null, mapDispatchToProps)(EditTaskContainer)