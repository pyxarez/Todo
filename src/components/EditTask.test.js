import React from 'react';
import { shallow } from 'enzyme';

import EditTask from './EditTask';

const setup = () => {
  const props = {
    onSave: () => {},
    onCancel: () => {}
  };

  const wrapper = shallow(
    <EditTask {...props}>
      <div className='test-class'/>
    </EditTask>
  );

  return {
    props,
    wrapper
  }
};

describe('Components :: EditTask', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render children', () => {
    const { props } = setup();
    const wrapper = shallow(<EditTask {...props} ></EditTask>);
    
    expect(EditTask).toMatchSnapshot();
  });
});