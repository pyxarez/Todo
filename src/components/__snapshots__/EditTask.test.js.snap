exports[`Components :: EditTask should not render children 1`] = `[Function]`;

exports[`Components :: EditTask should render self and subcomponents 1`] = `
<div
  className="my-editTask-component">
  <div
    className="buttons-container">
    <button
      onClick={[Function]}>
      Save changes
    </button>
    <button
      onClick={[Function]}>
      Cancel
    </button>
  </div>
  <div
    className="test-class" />
</div>
`;
